File: exercism/javascript/complete/roman-numerals/roman-numerals.js
const toRoman = (number) => {
  let romans = {"M":1000,"CM":900,"D":500,"CD":400,"C":100,"XC":90,"L":50,"XL":40,"X":10,"IX":9,"V":5,"IV":4,"I":1};
  let numerals = "";
  for (let key in romans) {
    let temp = Math.floor(number / romans[key]);
    if (temp > 0) {
      numerals += key.repeat(temp);
      number -= romans[key]*temp;
    }
  }
  return numerals;
}

export default toRoman;

File: exercism/ruby/complete/roman-numerals/roman_numerals.rb
class Integer
  def to_roman
    roman_string = ""
    num = self
    numerals = {1=>'I',5=>'V',10=>'X',50=>'L',100=>'C',500=>'D',1000=>'M'}
    numerals.sort_by{|k,v|-k}.each do |number, letter|
      if num >= number
        roman_string += letter * (num / number)
        num %= number
      elsif num >= number - number / 10 && number.to_s[0] == '1'
        roman_string += numerals[number/10] + letter
        num -= number - number/10
      elsif num >= number - number / 5 && number.to_s[0] == '5'
        roman_string += numerals[number/5] + letter
        num -= number - number/5
      end
    end
    roman_string
  end
end
